nav2_motion_primitives/include/nav2_motion_primitives/motion_primitive.hpp

```diff
@@ -24,127 +24,137 @@
 #include "rclcpp/rclcpp.hpp"
 #include "geometry_msgs/msg/twist.hpp"
-#include "nav2_tasks/task_status.hpp"
-#include "nav2_tasks/task_server.hpp"
+#include "nav2_util/simple_action_server.hpp"
 #include "nav2_robot/robot.hpp"
 
 namespace nav2_motion_primitives
 {
 
-using namespace std::chrono_literals;  //NOLINT
+enum class Status : int8_t
+{
+  SUCCEEDED = 1,
+  FAILED = 2,
+  RUNNING = 3,
+};
 
-template<class CommandMsg, class ResultMsg>
-const char * getTaskName();
+using namespace std::chrono_literals;  //NOLINT
 
-template<class CommandMsg, class ResultMsg>
+template<typename ActionT>
 class MotionPrimitive
 {
 public:
-  explicit MotionPrimitive(rclcpp::Node::SharedPtr & node)
+  using GoalHandle = rclcpp_action::ServerGoalHandle<ActionT>;
+  using ActionServer = nav2_util::SimpleActionServer<ActionT>;
+
+  explicit MotionPrimitive(rclcpp::Node::SharedPtr & node, const std::string & primitive_name)
   : node_(node),
-    task_server_(nullptr),
-    taskName_(nav2_tasks::getTaskName<CommandMsg, ResultMsg>())
+    primitive_name_(primitive_name),
+    action_server_(nullptr)
   {
-    robot_ = std::make_unique<nav2_robot::Robot>(node);
-
-    task_server_ = std::make_unique<nav2_tasks::TaskServer<CommandMsg, ResultMsg>>(node, false);
-
-    task_server_->setExecuteCallback(std::bind(&MotionPrimitive::run, this, std::placeholders::_1));
-
-    // Start listening for incoming Spin task requests
-    task_server_->start();
-
-    RCLCPP_INFO(node_->get_logger(), "Initialized the %s server", taskName_.c_str());
+    configure();
   }
 
   virtual ~MotionPrimitive()
   {
-    task_server_->stop();
+    cleanup();
   }
 
   // Derived classes can override this method to catch the command and perform some checks
   // before getting into the main loop. The method will only be called
   // once and should return SUCCEEDED otherwise behavior will return FAILED.
-  virtual nav2_tasks::TaskStatus onRun(
-    const typename /*nav2_tasks::*/ CommandMsg::SharedPtr command) = 0;
+  virtual Status onRun(const std::shared_ptr<const typename ActionT::Goal> command) = 0;
+
 
   // This is the method derived classes should mainly implement
   // and will be called cyclically while it returns RUNNING.
   // Implement the behavior such that it runs some unit of work on each call
   // and provides a status.
-  virtual nav2_tasks::TaskStatus onCycleUpdate(
-    /*nav2_tasks::*/ ResultMsg & result) = 0;
+  virtual Status onCycleUpdate() = 0;
 
-  // Runs the behavior
-  nav2_tasks::TaskStatus run(
-    const typename /*nav2_tasks::*/ CommandMsg::SharedPtr command)
-  {
-    RCLCPP_INFO(node_->get_logger(), "%s attempting behavior", taskName_.c_str());
+protected:
+  rclcpp::Node::SharedPtr node_;
+  std::string primitive_name_;
+  std::shared_ptr<nav2_robot::Robot> robot_;
+  std::unique_ptr<ActionServer> action_server_;
 
-    auto status = onRun(command);
+  void configure()
+  {
+    RCLCPP_INFO(node_->get_logger(), "Configuring ", primitive_name_.c_str());
```

Comment:
There is no "%s" in the string to show the primitive name.

Reply:
Fixed